api/server/router/network/network_routes.go

```diff
@@ -79,8 +80,16 @@ func (n *networkRouter) postNetworkCreate(ctx context.Context, w http.ResponseWr
 		return err
 	}
 
-	if _, err := n.clusterProvider.GetNetwork(create.Name); err == nil {
-		return libnetwork.NetworkNameError(create.Name)
+	if nw, err := n.clusterProvider.GetNetwork(create.Name); err == nil {
```

Comment:
I agree with @AkihiroSuda. It's ridiculous not to allow creating a network if its name is a prefix of another network's ID.

And using longer names in the test won't solve the problem, it will just make the failures less likely - so when they do happen, they will be very hard to reproduce and cause a lot of confusion.